server/user-bootstrap/shared-utils.js

```diff
@@ -52,7 +57,8 @@ module.exports = {
 			];
 
 		allowedKeys.forEach( function( key ) {
-			user[ key ] = obj[ key ];
+			// Decode if value is not falsey
+			user[ key ] = obj[ key ] ? decodeEntities( obj[ key ] ) : obj[ key ];
```

Comment:
we could write this `user[ key ] = obj[ key ] && decodeEntities( obj[ key ] )` :)